File: GatewayBackend/Data/Models/ContentPage.cs

namespace Data.Models
{
	public class ContentPage
	{
		public int Id { get; set; }
		public string Slug { get; set; } = null!;
		public string Title { get; set; } = null!;
		public string Area { get; set; } = null!;
		public string? Description { get; set; }
		public bool ComingSoon { get; set; } = false;
		public int SortOrder { get; set; } = 0;
		public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
		public DateTime? UpdatedAt { get; set; }
		public virtual ICollection<ContentSection> Sections { get; set; } = new List<ContentSection>();
	}
}

File: GatewayUI/react-ui/src/app/models/contentPage.ts
export interface ContentPageSummary {
	id: number;
	slug: string;
	title: string;
	area: string;
	description: string | null;
	comingSoon: boolean;
	sortOrder: number;
}

export interface ContentSection {
	id: number;
	title: string | null;
	body: string;
	sortOrder: number;
}

export interface ContentPageDetail {
	id: number;
	slug: string;
	title: string;
	area: string;
	description: string | null;
	comingSoon: boolean;
	sortOrder: number;
	sections: ContentSection[];
}

export interface ContentPageToCreate {
	slug: string;
	title: string;
	area: string;
	description?: string;
	comingSoon: boolean;
	sortOrder: number;
	sections: ContentSectionToCreate[];
}

export interface ContentSectionToCreate {
	title?: string;
	body: string;
	sortOrder: number;
}

export interface ContentPageToUpdate {
	title?: string;
	description?: string;
	comingSoon?: boolean;
	sortOrder?: number;
	area?: string;
}

export interface ContentSectionToUpdate {
	title?: string;
	body?: string;
	sortOrder?: number;
}

export interface ReorderSections {
	sectionIds: number[];
}

export interface ReorderPages {
	pageIds: number[];
}
